x/errors/errors.go

```diff
@@ -106,3 +108,54 @@ func GetInnerInvalidParamsError(err error) error {
 	}
 	return nil
 }
+
+// MultiError is an immutable error that packages a list of errors.
```

Comment:
Immutable is nice, but I wonder if `multiErr.Add(err)` is considerably nicer than `multiErr = multiErr.Add(err)`?  I can see someone easily forgetting to reassign their variable and missing adding errors.